state/virtualhostkeys.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/mgo/v3"
+	"github.com/juju/mgo/v3/txn"
+)
+
+// unitHostKeyID provides the virtual host key
+// lookup value for a unit based on the unit name.
+func unitHostKeyID(unitName string) string {
+	return "unit" + "-" + unitName + "-" + "hostkey"
+}
+
+// machineHostKeyID provides the virtual host key
+// lookup value for a machine based on the machine ID.
+func machineHostKeyID(machineID string) string {
+	return "machine" + "-" + machineID + "-" + "hostkey"
+}
+
+// VirtualHostKey represents the state of a virtual host key.
+type VirtualHostKey struct {
+	doc virtualHostKeyDoc
```

Comment:
What purpose does having the entire doc here have?
We just want to return the host key content right? 
Did we consider
```
type VirtualHostKey []byte
func (st *State) MachineVirtualHostKey(machineID string) (*VirtualHostKey, error)
```
Maybe there's even a core `type VirtualHostKey []byte` already that can be used.

ie there's no need for the struct

Reply:
You're right in that the doc in its entirety is not needed, it's only used by the getter for returning the host key.
But, at least it's consistent with other state structs. I don't think this change nets much, if anything.

Reply:
I'll leave it as is. If there is a strong reason to change it, I'll tweak it in a follow-up.